cosmosdb/cosmosdb_data_mappings.bal

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+function convertToDatabaseResponse(json jsonResponse) returns DatabaseResponse {
+    DatabaseResponse databaseResponse = {};
+    databaseResponse.database = convertToDatabase(jsonResponse);
+    return databaseResponse;
+}
+
+function convertToDatabase(json jsonResponse) returns Database {
+    Database database = {};
+    database.id = jsonResponse.id.toString();
+    database.rid = jsonResponse._rid.toString();
+    return database;
+}
+
+function convertToDatabaseListResponse(json jsonResponse) returns DatabaseListResponse {
+    DatabaseListResponse databaseListResponse = {};
+    if (jsonResponse.Databases != ()) {
+        json[] jsonDatabases = <json[]>jsonResponse.Databases;
+        int i = 0;
+        foreach json jsonDatabase in jsonDatabases {
+            databaseListResponse.databases[i] = convertToDatabase(jsonDatabase);
+            i = i + 1;
+        }
+    }
+    return databaseListResponse;
+}
+
+function convertToCollectionResponse(json jsonResponse) returns CollectionResponse {
+    CollectionResponse collectionResponse = {};
+    collectionResponse.collection = convertToDatabase(jsonResponse);
+    return collectionResponse;
+}
+
+function convertToCollection(json jsonResponse) returns Collection {
```

Comment:
Is there a reason to introduce functions to manually convert a `json` to a `record` instead of converting `json` to `record` using the `.convert()` method? Do we have additional fields in the `json` we are omitting?

Reply:
yes. During the design review it was decided to send only return the mostly required parameters instead of return the whole parameters.